source/s3_request_messages.c

```diff
@@ -199,6 +223,10 @@ struct aws_http_message *aws_s3_complete_multipart_message_new(
 
     aws_http_headers_erase(headers, g_content_type_header_name);
 
+    if (aws_http_headers_has(headers, g_content_md5_header_name)) {
+        aws_http_headers_erase(headers, g_content_md5_header_name);
```

Comment:
Should I have
```c++
if (aws_http_headers_erase(headers, g_content_md5_header_name)) {
    goto error_clean_up;
}
```
here?

Is there any reason why we don't check the result of `aws_http_headers_erase(headers, g_content_length_header_name)`?

Reply:
Because the header not being found returns an error and then it early outs in valid cases.  I guess it might be more correct for us to check the error code and early out if it's != HEADER_NOT_FOUND.

Reply:
You could could just assert that if there’s an error code it’s because the header wasn’t found